core/src/commons/Data.cpp

```diff
@@ -160,11 +158,11 @@ void Data::set_weight_index(size_t index) {
   disallowed_split_variables.insert(index);
 }
 
-void Data::get_all_values(std::vector<double>& all_values,
+void Data::get_all_values(size_t& num_unique_samples,
```

Comment:
I've never seen a `size_t` being passed as a _reference_, in particular as non-const. I imagine the reason for this choice was that it allows us  keep this function's return type as `void`. Could we instead, e.g., return a struct with all the information we want?

Reply:
Looks like idiomatic C++ to me. Maybe I'm stuck in the early 00s.

Reply:
Yes, it's to keep it consistent with the rest of grf, like `find_best_split_value`